reactor-netty-http/src/test/java/reactor/netty/http/Http2Tests.java

```diff
@@ -479,7 +480,21 @@ private void doTestMonoRequestBodySentAsFullRequest(Publisher<? extends ByteBuf>
 	}
 
 	@Test
-	void testIssue1394_HttpEndpointH2Protocol() {
+	void testIssue1394_SchemeHttp() {
+		// When only H2C is specified, "prior-knowledge" is used and stream id is 3
+		doTestIssue1394_SchemeHttp("3", HttpProtocol.H2C);
+		doTestIssue1394_SchemeHttp("3", HttpProtocol.H2, HttpProtocol.H2C);
+
+		// When only HTTP11 is specified, "Upgrade" header is used and stream id is 1
```

Comment:
the comment should read "when both HTTP11 and H2C are specified"

Reply:
isn't this a case where H2C was expected? or does the streamid 1 indicate H2C was selected ? (sorry I'm not familiar with the protocol and concepts of prior knowledge vs upgrade header). My feeling is that the implementation and the tests are hard to correlate with the matrix from the discussion, and would benefit from an isolated utility method that plainly returns the selected protocol, given a scheme and an array of protocols as input. wdyt?

Reply:
comments are updated